public/app/features/plugins/extensions/usePluginComponent.tsx

```diff
@@ -16,11 +17,19 @@ export function usePluginComponent<Props extends object = {}>(id: string): UsePl
   const registry = useExposedComponentsRegistry();
   const registryState = useObservable(registry.asObservable());
   const pluginContext = usePluginContext();
+  const { isLoading: isLoadingAppPlugins } = useLoadAppPlugins(getExposedComponentPluginDependencies(id));
```

Comment:
Q: do we need to handle an error case here as well?

Reply:
Hmm, it's a good question! I was not doing it here because the [`preload()` function in `pluginPreloader.ts` ](https://github.com/grafana/grafana/blob/61ab73d72a077069c752d4811dea5525c66a05fc/public/app/features/plugins/pluginPreloader.ts#L38) is handling the errors, and as far as I understand there _shouldn't_ be any errors that make their way up there (in the general scenarios). I was kinda imagining this `useLoadAppPlugins()` hook like something that you can just call, and in your UI component you only need to care about if it's loading or not, everything else would be handled inside. What do you think?

Reply:
Yeah, just had a look at the underlying code and it looks good!